test/lib/test-stack.ts

```diff
@@ -61,6 +64,77 @@ export class TestStack extends cdk.Stack {
       },
     });
 
+    /**
+     * Distributor example. 
+     * 
+     * Requires a bucket to hold install/update/uninstall scripts.
+     */
+    const bucketName = `${cdk.Stack.of(this).account}-cdk-ssm-document-storage`;
+    const bucket = new s3.Bucket(this, 'DistributorPackages', {
+      bucketName: bucketName,
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      enforceSSL: true,
+      encryption: s3.BucketEncryption.KMS_MANAGED,
+      // Makes for easy destroy and rerun of this stack over and over.
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: true,
+    });
+    const packageDeploy = new s3deploy.BucketDeployment(this, 'distribution-packages', {
+      sources: [s3deploy.Source.asset('../test/documents/distributor')],
+      destinationBucket: bucket
+    });
+
+    file = path.join(
+      __dirname,
+      '../documents/distributor/v1/manifest.json'
+    );
+    const docE = new Document(this, `SSM-Distribution-Package`, {
+      documentType: 'Package',
+      name: 'Test-Distributon-Package',
+      content: fs.readFileSync(file).toString(),
+      versionName: '1.0-Custom-Name',
+      attachments: [
+        { key: "SourceUrl", values: [`s3://${bucketName}/v1`] }
+      ]
+    });
+
+    // Comment `docE` above and uncommet this to simulate an update. 
+    // file = path.join(
+    //   __dirname,
+    //   '../documents/distributor/v2/manifest.json'
+    // );
+    // const docE = new Document(this, `SSM-Distribution-Package`, {
+    //   documentType: 'Package',
+    //   name: 'Test-Distributon-Package',
+    //   content: fs.readFileSync(file).toString(),
+    //   versionName: '2.0-Better-Than_Sliced_Bread',
+    //   attachments: [
+    //     { key: "SourceUrl", values: [`s3://${bucketName}/v2`] }
+    //   ]
+    // });
```

Comment:
As it states this is how I tested update. Comment out the `docE` and then uncomment the follow chunk that is commented out.